File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/systemframework/corepatch/ReturnConstant.java
package com.sevtinge.cemiuiler.module.hook.systemframework.corepatch;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XSharedPreferences;

public class ReturnConstant extends XC_MethodHook {
    private final XSharedPreferences prefs;
    private final String prefsKey;
    private final Object value;

    public ReturnConstant(XSharedPreferences prefs, String prefsKey, Object value) {
        this.prefs = prefs;
        this.prefsKey = prefsKey;
        this.value = value;
    }

    @Override
    protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
        super.beforeHookedMethod(param);
        prefs.reload();
        if (prefs.getBoolean(prefsKey, true)) {
            param.setResult(value);
        }
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/systemframework/corepatch/ReturnConstant.kt
package com.sevtinge.cemiuiler.module.systemframework.corepatch

import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.XSharedPreferences


open class ReturnConstant(
    private val prefs: XSharedPreferences,
    private val prefsKey: String,
    private val value: Any?
) :
    XC_MethodHook() {
    @Throws(Throwable::class)
    override fun beforeHookedMethod(param: MethodHookParam) {
        super.beforeHookedMethod(param)
        prefs.reload()
        if (prefs.getBoolean(prefsKey, false)) {
            param.result = value
        }
    }
}